### DISCUSSION TITLE
Axios gives an error after building as a library TypeError: (0 , _axios.default) is not a function

### DISCUSSION THREAD

**Discussion by Dmitriy-Krivetsky** [axios/axios#5745] [Q&A]
Axios version: 1.4.0
I have two packages (core and my package)
I make requests in the core package
In core package requests work, as soon as I make a build and run in my package I get an error:
`TypeError: (0 , _axios.default) is not a function`

The package is assembled without minify
`rm -rf dist && NODE_ENV=production babel src/lib --out-dir dist --copy-files`

I've tried a build with minify via vite - it works, but at the moment it's not the right solution for us

Downgrading to 1.1.2 also had no effect 

Example code:
```javascript
import axios from 'axios';

const getAPIDevices = async ({ prefUrl, token, siteId }) =>
  axios.get(`${prefUrl}/structure/${siteId}`, {
    headers: {
      Authorization: `Bearer ${token}`,
    }
  });
```

**Comment by TrusterZero** [comment]
> Axios version: 1.4.0 I have two packages (core and my package) I make requests in the core package In core package requests work, as soon as I make a build and run in my package I get an error: `TypeError: (0 , _axios.default) is not a function`
> 
> The package is assembled without minify `rm -rf dist && NODE_ENV=production babel src/lib --out-dir dist --copy-files`
> 
> I've tried a build with minify via vite - it works, but at the moment it's not the right solution for us
> 
> Downgrading to 1.1.2 also had no effect
> 
> Example code:
> 
> ```js
> import axios from 'axios';
> 
> const getAPIDevices = async ({ prefUrl, token, siteId }) =>
>   axios.get(`${prefUrl}/structure/${siteId}`, {
>     headers: {
>       Authorization: `Bearer ${token}`,
>     }
>   });
> ```

Did you ever solve it?

**Comment by jasonsaayman** [comment]
This usually points to a module interop/build issue rather than Axios itself.

Axios v1 is published as ESM with conditional exports, and Babel transpiling only your source files can sometimes produce code that expects a `.default` export shape that the consuming package/runtime does not actually provide.

Things to check:

```js
// If your built output is CommonJS, try:
const axios = require('axios');
```

or enable Babel/TypeScript interop consistently, for example `esModuleInterop` / `allowSyntheticDefaultImports` on the TypeScript side, or make sure Babel is not rewriting ESM imports in a way your consuming package cannot execute.

Also make sure `axios` is declared as a dependency or peer dependency of the package that uses it, not only of the consuming app.

A practical workaround is to avoid publishing half-transpiled module output: either keep ESM output as ESM, or build a proper CJS bundle that handles interop consistently.
